src/impl/device/rotarysensor/IMU.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * @author Alex Riensche, UNL
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#include "okapi/impl/device/rotarysensor/IMU.hpp"
+#include "okapi/api/util/mathUtil.hpp"
+
+namespace okapi {
+IMU::IMU(std::uint8_t iport, IMUAxes iaxis):port(iport),axis(iaxis) {
+}
+
+IMU::~IMU() = default;
+
+double IMU::get() const {
+  pros::c::euler_s_t eu = pros::c::imu_get_euler(port);
+  static double angle = 0;
+  switch(axis) {
+    case IMUAxes::x:
+    angle = eu.roll;
+    break;
+    case IMUAxes::y:
+    angle = eu.pitch;
+    break;
+    case IMUAxes::z:
+    angle = eu.yaw;
+    break;
+  }
+  angle += offset;
+  if(angle > 180 || angle < -180) {
+    angle += -360 * (offset/std::abs(offset));
+  }
+  return angle;
+}
+
+double IMU::getRemapped(const double iupperBound, const double ilowerBound) const {
+  double value = get();
+
+  return remapRange(value, -180, 180, ilowerBound, iupperBound);
+}
+
+double IMU::getAcc() {
+  pros::c::imu_accel_s_t accel = pros::c::imu_get_accel(port);
+  switch(axis) {
+    case IMUAxes::x:
+    return accel.x;
+    break;
+    case IMUAxes::y:
+    return accel.y;
+    break;
+    case IMUAxes::z:
+    return accel.z;
+    break;
+  }
+  return -1;
+}
+
+std::int32_t IMU::reset() {
+  const auto value = get() - offset;
+  offset = -1 * value;
```

Comment:
This is a weird way to do the offset. I would rather you did `offset = get()`.

Reply:
In order to assign a new offset, the old offset must be removed. To implement this, I will first zero offset before running ```offset = get();```.

Reply:
Yeah, you are right; what I meant was more like `offset = <get the axis value from PROS>` instead of `offset = get()`.

Reply:
That can be done. Take a look at my last commit. Would you like me to use the pros get in a switch or keep it as-is?

Reply:
I think it would be good to pull this out into a method
```
  pros::c::euler_s_t eu = pros::c::imu_get_euler(port);
  static double angle = 0;
  switch (axis) {
  case IMUAxes::x:
    angle = eu.roll;
    break;
  case IMUAxes::y:
    angle = eu.pitch;
    break;
  case IMUAxes::z:
    angle = eu.yaw;
    break;
  }
```
and use it to get the offset. Also, you don't need to have `angle` be `static`.

Reply:
I have added a private `readAngle()` method, and implemented it in `get()` and `calibrate()`. The commit compiles, however I want to test it on my robot later tonight.

Reply:
Just tested my latest pull request, the new private member class works as expected.